File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.swift
import Foundation

func solution(_ array:[Int]) -> Int {
    var cnt:[Int:Int] = [:]
    for a in array { cnt[a] = 0 }
    
    for a in array { cnt[a] = cnt[a]! + 1 }
    
    var max_cnt = 0
    for k in cnt.keys { if cnt[k]! > max_cnt { max_cnt = cnt[k]! } }
    
    var max:[Int] = []
    for k in cnt.keys { if cnt[k]! == max_cnt { max.append(k) } }
    
    return max.count > 1 ? -1 : max[0]
}

File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.java
import java.util.*;

class Solution {
    public int solution(int[] array) {
        Map<Integer, Integer> map = new HashMap<>();
        for(int num : array) {
            map.put(num, map.getOrDefault(num, 0)+1);
        }
        
        int max = 0;
        int max_num = 0;
        for(int key : map.keySet()) {
            if(max < map.get(key)) {
                max = map.get(key);
                max_num = key;
            }
        }
        
        int max_num_cnt = 0;
        for(int key : map.keySet()) {
            if(map.get(key) == max) max_num_cnt++;
        }
        
        if(max_num_cnt == 1) return max_num;
        else return -1;
    }
}